packages/smooth_app/lib/bottom_sheet_views/user_preferences_view.dart

```diff
@@ -22,143 +26,232 @@ class UserPreferencesView extends StatelessWidget {
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+            Column(
+              mainAxisSize: MainAxisSize.max,
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: <Widget>[
+                ClipRect(
+                  child: BackdropFilter(
+                    filter: ImageFilter.blur(
+                      sigmaX: 4.0,
+                      sigmaY: 4.0,
+                    ),
+                    child: Container(
+                      color: Colors.black12,
+                      padding: const EdgeInsets.symmetric(
+                          horizontal: 12.0, vertical: 20.0),
+                      child: SmoothMainButton(
+                        text: 'OK',
+                        onPressed: () {
+                          Navigator.pop(context);
+                          if (callback != null) {
+                            callback();
+                          }
+                        },
                       ),
                     ),
                   ),
-                ],
-              ),
-            ],
-          ),
+                ),
+              ],
+            ),
+          ],
         ),
       ),
     );
   }
 
-  Widget _generatePreferenceRow(String variable) {
+  Widget _generatePreferenceRow(
+    final Attribute variable,
+    final double screenWidth,
+    final UserPreferences userPreferences,
+    final UserPreferencesModel userPreferencesModel,
+  ) {
+    final double iconWidth =
+        (screenWidth - _TYPICAL_PADDING_OR_MARGIN * 5) * _PCT_ICON;
+    final double sliderWidth =
+        (screenWidth - _TYPICAL_PADDING_OR_MARGIN * 5) * (1 - _PCT_ICON);
+    final PreferencesValue importance =
+        userPreferencesModel.getPreferencesValue(
+      variable.id,
+      userPreferences,
+    );
     return Container(
-      padding: const EdgeInsets.all(10.0),
-      margin: const EdgeInsets.all(10.0),
+      padding: const EdgeInsets.all(_TYPICAL_PADDING_OR_MARGIN),
+      margin: const EdgeInsets.all(_TYPICAL_PADDING_OR_MARGIN),
+      width: screenWidth,
       decoration: BoxDecoration(
           color: Colors.black.withAlpha(5),
           borderRadius: const BorderRadius.all(Radius.circular(20.0))),
-      child: Column(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.center,
         children: <Widget>[
-          Row(
-            mainAxisSize: MainAxisSize.max,
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: <Widget>[
-              Text(UserPreferencesModel.getVariableName(variable)),
-            ],
+          Container(
+            width: iconWidth,
+            child: SvgPicture.network(
+              variable.iconUrl,
+              alignment: Alignment.centerLeft,
+              width: iconWidth,
+              fit: BoxFit.scaleDown,
```

Comment:
This results in this:

![image](https://user-images.githubusercontent.com/8158668/103180257-762dba00-4894-11eb-9e98-ea3a64e716a6.png)

I think it should be BoxFit.contains, so that the icons for allergens can fill the container.

But I don't understand why the Nutri-Score icon is not shown completely, that's very strange. Maybe a bug in flutter_svg, it seems to have many open issues.

Reply:
OK I did a few tests, and it seems that it is indeed a bug with flutter_svg handling of the viewbox of the SVG when it's different than the width and height.
Don't worry about it, I'll just change the SVG files so that the viewbox matches the actual width and height.

Reply:
Server side fix: https://github.com/openfoodfacts/openfoodfacts-server/issues/4674